presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogFilter.java

```diff
@@ -0,0 +1,288 @@
+ * non-matching lines as part of the entry on the previous line.
+ *
+ * @see Builder
+ */
+public class LogFilter
+{
+    private final Path file;
+    private final Pattern logPattern;
+    private final String lineSeparator;
+    private final ImmutableMap<String, Predicate<String>> namedGroupFilters;
+    private final ImmutableMap<Integer, Predicate<String>> numberedGroupFilters;
+    private final Integer maxEntries;
+    private final boolean keepFirst;
+
+    private LogFilter(Path file, Pattern logPattern, String lineSeparator,
+            ImmutableMap<String, Predicate<String>> namedGroupFilters,
+            ImmutableMap<Integer, Predicate<String>> numberedGroupFilters,
+            Integer maxEntries, boolean keepFirst)
+            throws FileNotFoundException
+    {
+        this.file = requireNonNull(file);
+        this.logPattern = requireNonNull(logPattern);
+        this.lineSeparator = requireNonNull(lineSeparator);
+        this.namedGroupFilters = requireNonNull(namedGroupFilters);
+        this.numberedGroupFilters = requireNonNull(numberedGroupFilters);
+        this.maxEntries = maxEntries; // null -> no maximum
+        this.keepFirst = keepFirst;
+        if (!Files.exists(file)) {
+            throw new FileNotFoundException(file.toString());
+        }
+    }
+
+    public Stream<String> streamEntries()
+            throws IOException, ConcurrentModificationException
+    {
+        LogEntries logEntryBuilders = Files.lines(file).collect(
+                LogEntries::new,
+                LogEntries::addLine,
+                (a, b) -> {
+                    throw new ConcurrentModificationException();
+                });
+        return logEntryBuilders.streamEntries().map(StringJoiner::toString);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    /**
+     * A builder for creating {@link LogFilter LogFilters}.
+     *
+     * Builder instances can be safely reused.
+     */
+    public static class Builder
+    {
+        private Path file;
+        private Pattern logPattern;
+        private String lineSeparator = System.getProperty("line.separator");
+        private ImmutableMap.Builder<String, Predicate<String>> namedGroupFilters = ImmutableMap.builder();
+        private ImmutableMap.Builder<Integer, Predicate<String>> numberedGroupFilters = ImmutableMap.builder();
+        private Integer maxEntries;
+        private boolean keepFirst = false;
+
+        private Builder() {}
+
+        public LogFilter build()
+                throws FileNotFoundException
+        {
+            return new LogFilter(file, logPattern, lineSeparator,
+                    namedGroupFilters.build(),
+                    numberedGroupFilters.build(),
+                    maxEntries, keepFirst);
+        }
+
+        public Builder setFile(Path file)
+        {
+            requireNonNull(file);
+            this.file = file;
```

Comment:
I think in the rest of code you used to use `this.X = requireNonNull(x, "foo")` pattern.